slm_porosity/heat_transfer.f90

```diff
@@ -398,28 +399,33 @@ FUNCTION user_Drhs_diag (meth)
     IMPLICIT NONE
     INTEGER, INTENT (IN) :: meth
     REAL (pr), DIMENSION (n) :: user_Drhs_diag
-    REAL (pr), DIMENSION (ng) :: T
+    REAL (pr), DIMENSION (ng) :: T, phi, Dphi, DT, k, c_p, Dk, dc_p
     INTEGER :: ie, shift, i
-    REAL (pr) :: first_der
     REAL (pr), DIMENSION (ng,dim) :: du, d2u, part1, part2
     REAL (pr), DIMENSION (ne,ng,dim) :: du_prev, du_dummy
 
     ie = n_var_enthalpy
     shift = ng*(ie-1)
-    first_der = conductivity_der - 2*capacity_der
     IF (IMEXswitch.LE.0) THEN
        user_Drhs_diag(shift+1:shift+ng) = 1.0_pr
     END IF
     IF (IMEXswitch .GE. 0) THEN
+       T = temperature(u_prev_timestep(shift+1:shift+ng))
+       phi = liquid_fraction(u_prev_timestep(shift+1:shift+ng))
+       Dphi = liquid_fraction(u_prev_timestep(shift+1:shift+ng), .TRUE.)
+       DT = Dtemperature(u_prev_timestep(shift+1:shift+ng),T)
```

Comment:
missed whitespace. You can check it by `egrep ',\w+' heat_transfer.f90`. Maybe, some other PR should be devoted to cleaning the code. It is up to you.